File: minji/week1/10866.py
'''
    [BOJ] 10866 덱
'''
import sys

N = int(sys.stdin.readline().rstrip())
deq = []

for _ in range(N):
    cmd = sys.stdin.readline().rstrip().split()
    if cmd[0] == "push_front":
        deq.insert(0, cmd[1])
    elif cmd[0] == "push_back":
        deq.append(cmd[1])
    elif cmd[0] =="pop_front":
        if deq:
            print(deq.pop(0))
        else:
            print(-1)
    elif cmd[0] == "pop_back":
        if deq:
            print(deq.pop())
        else:
            print(-1)
    elif cmd[0] == "size":
        print(len(deq))
    elif cmd[0] == "empty":
        if deq:
            print(0)
        else:
            print(1)
    elif cmd[0] == "front":
        if deq:
            print(deq[0])
        else:
            print(-1)
    elif cmd[0] == "back":
        if deq:
            print(deq[-1])
        else:
            print(-1)

File: minji/week1/10866.cpp
/*
	[BOJ] 10866 µ¦
*/
#include <iostream>
#include <deque>
#define endl "\n"
using namespace std;

deque<int> dq;
int N;

int main() {
	ios_base::sync_with_stdio(false); cin.tie(0); cout.tie(0);
	cin >> N;

	while (N) {
		string command;

		cin >> command;

		if (command == "push_front") {
			int num;
			cin >> num;
			dq.push_front(num);
		}
		else if (command == "push_back") {
			int num;
			cin >> num;
			dq.push_back(num);
		}
		else if (command == "pop_front") {
			if (!dq.empty()) {
				cout << dq.front() << endl;
				dq.pop_front();
			}
			else {
				cout << "-1" << endl;
			}
		}
		else if (command == "pop_back") {
			if (!dq.empty()) {
				cout << dq.back() << endl;
				dq.pop_back();
			}
			else {
				cout << "-1" << endl;
			}
		}
		else if (command == "size") {
			cout << dq.size() << endl;
		}
		else if (command == "empty") {
			cout << dq.empty() << endl;
		}
		else if (command == "front") {
			if (dq.size() == 0) {
				cout << "-1" << endl;
			}
			else {
				cout << dq.front() << endl;
			}
		}
		else if (command == "back") {
			if (dq.size() == 0) {
				cout << "-1" << endl;
			}
			else {
				cout << dq.back() << endl;
			}
		}
		--N;
	}
	return 0;
}